File: frontend/src/domain/game.ts
export type Coordinates = { lat: number; lng: number }

export type EventType = 'history' | 'legend' | 'culture'

export type Guess = {
  year: number
  coordinates: Coordinates
  timedOut: boolean
  hintIds?: string[]
}

export type RoundResult = {
  guess: Guess
  yearError: number
  distanceKm: number
  score: number
  maximumScore: number
  timedOut: boolean
  hintPenalty: number
}

export type Hotspot = {
  id: string
  title: string
  titleTt?: string
  kind: 'time' | 'place' | 'context' | 'legend' | 'object' | 'story'
  imageUrl: string
  imageSourceUrl?: string
  imageCredit?: string
  description: string
  descriptionTt?: string
  yaw: number
  pitch: number
}

export type PanoramaMoment = {
  year: number
  title: string
  titleTt?: string
  description: string
  descriptionTt?: string
  panoramaUrl: string
}

export type EventReveal = {
  title: string
  titleTt?: string
  subtitle: string
  subtitleTt?: string
  year: number
  place: string
  placeTt?: string
  coordinates: Coordinates
  description: string
  descriptionTt?: string
  sourceTitle: string
  sourceUrl: string
  hotspots: Hotspot[]
  panoramaTimeline: PanoramaMoment[]
}

export type Round = {
  id: string
  number: number
  total: number
  eventType: EventType
  panoramaUrl: string
  deadline: string
  result?: RoundResult
  reveal?: EventReveal
  hotspots?: Hotspot[]
}

export type Game = {
  id: string
  status: 'in_progress' | 'completed'
  score: number
  maximumScore: number
  currentRound?: Round
  summary?: Round[]
}

File: backend/internal/domain/game.go
package domain

import (
	"errors"
	"time"
)

var (
	ErrNotFound       = errors.New("not found")
	ErrInvalidGuess   = errors.New("invalid guess")
	ErrRoundCompleted = errors.New("round already completed")
	ErrGuessRequired  = errors.New("submit a guess before continuing")
	ErrGameCompleted  = errors.New("game already completed")
)

type GameStatus string

const (
	GameInProgress GameStatus = "in_progress"
	GameCompleted  GameStatus = "completed"
)

type Guess struct {
	Year        int         `json:"year"`
	Coordinates Coordinates `json:"coordinates"`
	TimedOut    bool        `json:"timedOut"`
}

type RoundResult struct {
	Guess        Guess `json:"guess"`
	YearError    int   `json:"yearError"`
	DistanceKM   int   `json:"distanceKm"`
	Score        int   `json:"score"`
	MaximumScore int   `json:"maximumScore"`
	TimedOut     bool  `json:"timedOut"`
}

type Round struct {
	ID        string       `json:"id"`
	EventID   string       `json:"eventId"`
	Result    *RoundResult `json:"result,omitempty"`
	StartedAt time.Time    `json:"startedAt"`
}

type Game struct {
	ID           string     `json:"id"`
	Rounds       []Round    `json:"rounds"`
	CurrentRound int        `json:"currentRound"`
	Score        int        `json:"score"`
	Status       GameStatus `json:"status"`
	CreatedAt    time.Time  `json:"createdAt"`
}
